src/usr/local/captiveportal/index.php

```diff
@@ -157,8 +163,12 @@
 	captiveportal_logportalauth("unauthenticated", $clientmac, $clientip, "ACCEPT");
 	portal_allow($clientip, $clientmac, "unauthenticated");
 
-} elseif (isset($config['voucher'][$cpzone]['enable']) && $_POST['accept'] && $_POST['auth_voucher']) {
-	$voucher = trim($_POST['auth_voucher']);
+} elseif (isset($config['voucher'][$cpzone]['enable']) && (($_POST['accept'] && $_POST['auth_voucher']) || $_REQUEST['voucher'])) {
+	if (isset($_POST['auth_voucher'])) {
+		$voucher = trim($_POST['auth_voucher']);
+	} else {
+		$voucher = trim($_REQUEST['voucher']);
```

Comment:
Similar to above, we shouldn't accept a voucher as-is to change the user state when it is supplied via GET.

It could pre-fill the voucher in the login form, but it should not take the value and log the user in automatically.

Reply:
Fixed.
Now it pre-fill the voucher in the login form